Python/pylifecycle.c

```diff
@@ -2942,46 +2973,52 @@ _Py_DumpExtensionModules(int fd, PyInterpreterState *interp)
     int header = 1;
     Py_ssize_t count = 0;
     pos = 0;
-    while (PyDict_Next(modules, &pos, &key, &value)) {
-        if (!PyUnicode_Check(key)) {
-            continue;
-        }
-        if (!_PyModule_IsExtension(value)) {
-            continue;
-        }
-        // Use the module name from the sys.modules key,
-        // don't attempt to get the module object name.
-        if (stdlib_module_names != NULL) {
-            int is_stdlib_ext = 0;
-
-            Py_ssize_t i = 0;
-            PyObject *item;
-            Py_hash_t hash;
-            // if stdlib_module_names is not NULL, it is always a frozenset.
-            while (_PySet_NextEntry(stdlib_module_names, &i, &item, &hash)) {
-                if (PyUnicode_Check(item)
-                    && PyUnicode_Compare(key, item) == 0)
-                {
-                    is_stdlib_ext = 1;
-                    break;
-                }
+    if (acquire_dict_lock_for_dump(modules) == 0) {
```

Comment:
Rather than adding one indentation level, can you just return if you cannot acquire the lock?

And maybe add a comment, like: "If we cannot acquire the lock, just don't dump the list of extension modules."

Reply:
Yeah looks better, thank you for suggestion